tests/unit/core/utils/test_index.js

```diff
@@ -63,795 +64,810 @@ import {
+    it('treats undefined as 1', () => {
+      expect(parsePage(undefined)).toBe(1);
+    });
   });
-});
 
-describe('sanitizeUserHTML', () => {
-  const sanitize = (...args) => sanitizeUserHTML(...args).__html;
+  describe('sanitizeUserHTML', () => {
+    const sanitize = (...args) => sanitizeUserHTML(...args).__html;
 
-  it('converts new lines to breaks', () => {
-    expect(sanitize(`
+    it('converts new lines to breaks', () => {
+      expect(sanitize(`
       first
       second
     `).trim()).toEqual('<br>      first<br>      second<br>');
-  });
+    });
 
-  it('allows some tags', () => {
-    const customHtml =
-      '<b>check</b> <i>out</i> <a href="http://mysite">my site</a>';
-    expect(sanitize(customHtml)).toEqual(customHtml);
-  });
+    it('allows some tags', () => {
+      const customHtml =
+        '<b>check</b> <i>out</i> <a href="http://mysite">my site</a>';
+      expect(sanitize(customHtml)).toEqual(customHtml);
+    });
 
-  it('does not allow certain tags', () => {
-    expect(sanitize('<b>my add-on</b> <script>alert("does XSS")</script>'))
-      .toEqual('<b>my add-on</b> ');
-  });
+    it('does not allow certain tags', () => {
+      expect(sanitize('<b>my add-on</b> <script>alert("does XSS")</script>'))
+        .toEqual('<b>my add-on</b> ');
+    });
 
-  it('does nothing to null values', () => {
-    expect(sanitize(null)).toEqual('');
+    it('does nothing to null values', () => {
+      expect(sanitize(null)).toEqual('');
+    });
   });
-});
 
-describe('sanitizeHTML', () => {
-  it('does not change links', () => {
-    const html = '<a href="http://example.org">link</a>';
-    expect(sanitizeHTML(html, ['a'])).toEqual({
-      __html: html,
+  describe('sanitizeHTML', () => {
+    it('does not change links', () => {
+      const html = '<a href="http://example.org">link</a>';
+      expect(sanitizeHTML(html, ['a'])).toEqual({
+        __html: html,
+      });
+    });
+
+    // This is a built-in feature of dompurify.
+    it('removes `target` attribute on links', () => {
+      const html = '<a href="http://example.org" target="_blank">link</a>';
+      expect(sanitizeHTML(html, ['a'])).toEqual({
+        __html: '<a href="http://example.org">link</a>',
+      });
     });
   });
 
-  // This is a built-in feature of dompurify.
-  it('removes `target` attribute on links', () => {
-    const html = '<a href="http://example.org" target="_blank">link</a>';
-    expect(sanitizeHTML(html, ['a'])).toEqual({
-      __html: '<a href="http://example.org">link</a>',
+  describe('getRelativePath', () => {
+    it('returns a path relative to the project root directory', () => {
+      expect(getRelativePath('/path/to/src/foo/index.js'))
+        .toEqual('src/foo/index.js');
+    });
+
+    it('returns the given filename if `src` is not in it', () => {
+      const filename = 'tests/unit/core/utils/test_index.js';
+
+      expect(getRelativePath(filename)).toEqual(filename);
```

Comment:
I'd like to see an additional test that asserts how `getRelativePath('src/file.js')` results in `src/file.js` -- that is, no stripping.

Reply:
ah correct 👍